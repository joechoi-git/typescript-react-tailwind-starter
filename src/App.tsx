import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { BrowserRouter as Router, Routes, Route, Link } from "react-router-dom";

interface ArticleProps {
    title: string;
    content: string;
}

function Article({ title, content }: ArticleProps): JSX.Element {
    return (
        <div>
            <h2>{title}</h2>
            <p>{content}</p>
        </div>
    );
}

export default function App(): JSX.Element {
    return (
        <Router>
            <nav>
                <img src={logo} className="App-logo" alt="logo" />
                <ul>
                    <li>
                        <Link to="/">Home</Link>
                    </li>
                    <li>
                        <Link to="/about">About</Link>
                    </li>
                    <li>
                        <Link to="/users">Users</Link>
                    </li>
                </ul>
            </nav>

            <Routes>
                <Route
                    path="/"
                    element={<Article title={"Home"} content={"Welcome to React App!"} />}
                />
                <Route
                    path="/about"
                    element={<Article title={"About"} content={"I am super awesome!"} />}
                />
                <Route
                    path="/users"
                    element={<Article title={"Users"} content={"How many folks came to visit?"} />}
                />
            </Routes>
        </Router>
    );
}
