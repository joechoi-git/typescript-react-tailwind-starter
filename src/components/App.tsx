import React from "react";
import { BrowserRouter as Router, Routes, Route, Link } from "react-router-dom";
import logo from "../logo.svg";
import "./App.css";
import { Article } from "./Article";

export default function App(): React.JSX.Element {
    return (
        <Router>
            <nav>
                <img src={logo} className="App-logo" alt="logo" />
                <ul>
                    <li>
                        <Link to="/">Home</Link>
                    </li>
                    <li>
                        <Link to="/about">About</Link>
                    </li>
                    <li>
                        <Link to="/users">Users</Link>
                    </li>
                </ul>
            </nav>

            <Routes>
                <Route
                    path="/"
                    element={<Article title={"Home"} content={"Welcome to React App!"} />}
                />
                <Route
                    path="/about"
                    element={<Article title={"About"} content={"I am super awesome!"} />}
                />
                <Route
                    path="/users"
                    element={<Article title={"Users"} content={"How many folks came to visit?"} />}
                />
            </Routes>
        </Router>
    );
}
