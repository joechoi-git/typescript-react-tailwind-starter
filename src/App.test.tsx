import React from "react";
import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

test("navigates through all the router links", async () => {
    render(<App />);

    // Get the navigation element
    const nav = screen.getByRole("navigation");
    console.log("nav", nav.outerHTML);

    // Click on the Home link and check if the Home component is displayed
    const homeLink = within(nav).getByText(/home/i);
    userEvent.click(homeLink);
    await new Promise((r) => setTimeout(r, 100));
    console.log("home clicked");
    screen.debug();
    expect(screen.getByRole("heading", { name: /home/i })).toBeInTheDocument();

    // Click on the About link and check if the About component is displayed
    const aboutLink = within(nav).getByText(/about/i);
    userEvent.click(aboutLink);
    await new Promise((r) => setTimeout(r, 100));
    console.log("about clicked");
    screen.debug();
    expect(screen.getByRole("heading", { name: /about/i })).toBeInTheDocument();

    // Click on the Users link and check if the Users component is displayed
    const usersLink = within(nav).getByText(/users/i);
    userEvent.click(usersLink);
    await new Promise((r) => setTimeout(r, 100));
    console.log("users clicked");
    screen.debug();
    expect(screen.getByRole("heading", { name: /users/i })).toBeInTheDocument();
});
