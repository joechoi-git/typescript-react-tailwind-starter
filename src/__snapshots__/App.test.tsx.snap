// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <img
      alt="logo"
      class="App-logo"
      src="test-file-stub"
    />
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/users"
        >
          Users
        </a>
      </li>
    </ul>
  </nav>
  <header>
    <h1>
      Users
    </h1>
    <p>
      Published on 
      <time
        datetime="2024-06-28"
      >
        June 28, 2024
      </time>
       by 
      <span>
        Author Name
      </span>
    </p>
  </header>
  <article>
    <section>
      <p>
        How many folks came to visit?
      </p>
    </section>
  </article>
</DocumentFragment>
`;
